Corona-Warn-App/src/main/java/de/rki/coronawarnapp/submission/auto/AutoSubmission.kt

```diff
@@ -0,0 +1,144 @@
+package de.rki.coronawarnapp.submission.auto
+
+import androidx.work.BackoffPolicy
+import androidx.work.Constraints
+import androidx.work.ExistingPeriodicWorkPolicy
+import androidx.work.NetworkType
+import androidx.work.PeriodicWorkRequestBuilder
+import androidx.work.WorkManager
+import de.rki.coronawarnapp.submission.SubmissionSettings
+import de.rki.coronawarnapp.submission.task.SubmissionTask
+import de.rki.coronawarnapp.task.TaskController
+import de.rki.coronawarnapp.task.TaskInfo
+import de.rki.coronawarnapp.task.common.DefaultTaskRequest
+import de.rki.coronawarnapp.task.submitBlocking
+import de.rki.coronawarnapp.util.TimeStamper
+import de.rki.coronawarnapp.worker.BackgroundConstants
+import kotlinx.coroutines.flow.map
+import org.joda.time.Instant
+import timber.log.Timber
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class AutoSubmission @Inject constructor(
+    private val timeStamper: TimeStamper,
+    private val submissionSettings: SubmissionSettings,
+    private val workManager: WorkManager,
+    private val taskController: TaskController
+) {
+
+    private var currentMode: Mode = Mode.DISABLED
```

Comment:
`curentMode` isn't used (anymore), is it?

Reply:
True